### PR TITLE
Further tests improvements

### PR DESCRIPTION
**Pull request by DPeterK**
More test improvements in the spirit of #34, and with the same set of goals in mind.

This set of changes has potentially caused a bigger diff as I have changed the order of a number of test classes and also added a few test classes to cover over a few holes in coverage of unit function and methods.

### CODE REVIEW

File: cf_units/tests/test_unit.py [hunk 1]
```diff
@@ -522,63 +522,56 @@ def test_unit_power(self):
...
-    #
-    # test: copy method
-    #
-    def test_copy_pass_0(self):
+class Test_copy(unittest.TestCase):
+
+    def test_basic(self):
         u = Unit('joule')
-        self.assertEqual(copy.copy(u) == u, True)
+        self.assertTrue(copy.copy(u) == u)
```
Comment (pelson) [inline cf_units/tests/test_unit.py]: `self.assertEqual(copy.copy(u) == u, True)`
->
`self.assertEqual(copy.copy(u), u)`

**Comment by pelson** [comment]
A rebase + quick change of assertions => merge. :+1:

File: cf_units/tests/test_unit.py [hunk 2]
```diff
@@ -522,63 +522,56 @@ def test_unit_power(self):
...
-        self.assertEqual(u == v, False)
+        self.assertTrue(u == 'meter')
 
-    def test_eq_pass_1(self):
+    def test_equivalent_units(self):
         u = Unit('meter')
         v = Unit('m.s-1')
         w = Unit('hertz')
-        self.assertEqual(u == (v / w), True)
+        self.assertTrue(u == (v / w))
```
Comment (pelson) [inline cf_units/tests/test_unit.py]: `self.assertTrue(u == (v / w))` => `self.assertEqual(u, v / w)`

File: cf_units/tests/test_unit.py [hunk 3]
```diff
@@ -522,63 +522,56 @@ def test_unit_power(self):
...
 
-    def test_eq_pass_2(self):
+    def test_non_equivalent_units(self):
         u = Unit('meter')
-        self.assertEqual(u == 'meter', True)
+        v = Unit('amp')
+        self.assertFalse(u == v)
```
Comment (pelson) [inline cf_units/tests/test_unit.py]: assertNotEqual

File: cf_units/tests/test_unit.py [hunk 4]
```diff
@@ -586,19 +579,31 @@ def test_eq_cross_category(self):
...
         self.assertFalse(m == n)
-        self.assertFalse(u == n)
 
-    #
-    # test: __ne__ method
-    #
-    def test_neq_pass_0(self):
+    def test_unknown(self):
+        u = Unit('unknown')
+        self.assertTrue(u == 'unknown')
```
Comment (pelson) [inline cf_units/tests/test_unit.py]: ditto

File: cf_units/tests/test_unit.py [hunk 5]
```diff
@@ -586,19 +579,31 @@ def test_eq_cross_category(self):
...
+
+    def test_no_unit(self):
+        u = Unit('no_unit')
+        self.assertTrue(u == 'no_unit')
```
Comment (pelson) [inline cf_units/tests/test_unit.py]: ditto

**Comment by DPeterK** [comment]
Odd assertions in tests corrected. Rebased.

**Comment by pelson** [comment]
Just a 1 char change needed now :wink:

**Comment by DPeterK** [comment]
I _know_ I already fixed that trailing whitespace :unamused: Let's hope that fixes it!

**Pull request merged**